### PR TITLE
Rewrote compound types tour #742

### PR DESCRIPTION
**Pull request by mghildiy**

### CODE REVIEW

**Comment by mghildiy** [comment]
PR 'Rewrote compound types tour #742' can be closed.

File: _tour/self-types.md [hunk 1]
```diff
@@ -24,12 +22,12 @@ trait User {
 }
 
 trait Tweeter {
-  this: User =>  // reassign this
-  def tweet(tweetText: String) = println(s"$username: $tweetText")
+	this: User =>  // reassign this
```
Comment (gmethvin) [inline _tour/self-types.md]: Is there a reason this was changed from two spaces to a tab? I'd prefer keeping two-space indents.

File: _tour/traits.md [hunk 2]
```diff
@@ -36,6 +34,12 @@ Extending the `trait Iterator[A]` requires a type `A` and implementations of the
 ## Using traits
 Use the `extends` keyword to extend a trait. Then implement any abstract members of the trait using the `override` keyword:
 ```tut
+trait Iterator[A] {
+  def hasNext: Boolean
+  def next(): A
+}
+
```
Comment (jvican) [inline _tour/traits.md]: Could we remove this extra space? :smile:

File: _tour/traits.md [hunk 3]
```diff
@@ -50,13 +54,13 @@ class IntIterator(to: Int) extends Iterator[Int] {
 
 
 val iterator = new IntIterator(10)
-println(iterator.next())  // prints 0
-println(iterator.next())  // prints 1
+iterator.next()  // prints 0
```
Comment (jvican) [inline _tour/traits.md]: `// return 0` rather than `// prints 0`?

**Comment by mghildiy** [comment]
I have replaced the tab with two spaces.

**Comment by jvican** [comment]
@mghildiy If you're looking for more, we still need one more contributor in https://github.com/scala/docs.scala-lang/pull/746 :smile:.

**Comment by mghildiy** [comment]
Would pick #746.Any deadline for it?.

**Comment by jvican** [comment]
Absolutely not, take your time. That's one of the gems that can actually be enlightening to work on if you're new to Scala; a solid understanding of implicits is paramount to become a better Scala programmer..

**Comment by jvican** [comment]
Let me know when you've addressed these comments and we merge @mghildiy.

**Comment by mghildiy** [comment]
I have already made the change(Ttab to 2 spaces in self-types.md) as
suggested by @gmethvin and created the PR.

**Comment by jvican** [comment]
It seems you forgot to address my review comments though?

File: _tour/traits.md (hunk 2 above)
Comment (mghildiy) [reply _tour/traits.md]: I assume by extra space you mean line #41.

Comment (mghildiy) [reply _tour/traits.md]: or extra space in 38,39.

**Comment by mghildiy** [comment]
Changes done

**Comment by SethTisue** [comment]
thank you @mghildiy!

**Pull request merged**